torchmetrics/functional/classification/auc.py

```diff
@@ -35,7 +33,7 @@ def _auc_update(x: Tensor, y: Tensor) -> Tuple[Tensor, Tensor]:
 
 def _auc_compute(x: Tensor, y: Tensor, reorder: bool = False) -> Tensor:
     if reorder:
-        x, x_idx = _stable_1d_sort(x)
+        x, x_idx = torch.sort(x)
```

Comment:
`torch.sort` is not stable, meaning that you will get different result if evaluating on cpu vs gpu:
```python
import torch
x = torch.randint(5, (10,))
torch.sort(x)
#torch.return_types.sort(
#values=tensor([0, 0, 1, 1, 1, 2, 3, 4, 4, 4]),
#indices=tensor([0, 2, 3, 4, 7, 9, 8, 1, 5, 6]))
torch.sort(x.cuda())
#torch.return_types.sort(
#values=tensor([0, 0, 1, 1, 1, 2, 3, 4, 4, 4], device='cuda:0'),
#indices=tensor([0, 2, 7, 4, 3, 9, 8, 1, 5, 6], device='cuda:0'))
```
this is the reason that we implemented a stable sorting algorithm